ngraph/core/src/op/reduce_prod.cpp

```diff
@@ -75,8 +76,11 @@ bool op::v1::ReduceProd::evaluate(const HostTensorVector& outputs,
     NGRAPH_OP_SCOPE(v1_ReduceProd_evaluate);
     NGRAPH_CHECK(validate_host_tensor_vector(inputs, 2));
     NGRAPH_CHECK(validate_host_tensor_vector(outputs, 1));
-    return reduce_prod::evaluate_product(
-        inputs[0], outputs[0], get_reduction_axes(), get_keep_dims());
+
+    const auto reduction_axes = get_normalized_axes_from_tensor(
+        inputs[1], get_input_partial_shape(0).rank(), get_friendly_name());
```

Comment:
I just wonder why we are using `get_input_partial_shape(0)` here
while `arg->get_shape()` on HostTensor is called later anyway

Reply:
The helper requires a `Rank` object for normalization purposes.

Reply:
That is a good question. We could use `inputs[0]->get_partial_shape().rank()` instead. @tomdol what do you think?

Reply:
I think will be better to use partial shape from HostTensor, because you will use this memory in calculations.

Reply:
Applied suggestion